source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc

```diff
@@ -215,18 +216,35 @@ void RawHttpClientImpl::check(RequestCallbacks& callbacks,
     headers = Http::createHeaderMap<Http::RequestHeaderMapImpl>(lengthZeroHeader());
   }
 
-  for (const auto& header : http_request.headers()) {
-    const Http::LowerCaseString key{header.first};
+  if (config_->encodeRawHeaders()) {
+    for (const auto& header : http_request.header_map().headers()) {
+      const Http::LowerCaseString key{header.key()};
 
-    // Skip setting content-length header since it is already configured at initialization.
-    if (key == Http::Headers::get().ContentLength) {
-      continue;
+      // Skip setting content-length header since it is already configured at initialization.
+      if (key == Http::Headers::get().ContentLength) {
+        continue;
+      }
+
+      if (key == Http::Headers::get().Path && !config_->pathPrefix().empty()) {
+        headers->addCopy(key, absl::StrCat(config_->pathPrefix(), header.raw_value()));
```

Comment:
nit: this line isn't being covered by tests (https://storage.googleapis.com/envoy-pr/2ef95da/coverage/source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc.gcov.html), consider modifying/adding a test.